src/Symfony/Component/DependencyInjection/Compiler/AttributeAutoconfigurationPass.php

```diff
@@ -11,47 +11,103 @@
 
 namespace Symfony\Component\DependencyInjection\Compiler;
 
+use Symfony\Component\DependencyInjection\Argument\BoundArgument;
+use Symfony\Component\DependencyInjection\Argument\ServiceLocatorArgument;
+use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+use Symfony\Component\DependencyInjection\Attribute\BindTaggedIterator;
+use Symfony\Component\DependencyInjection\Attribute\BindTaggedLocator;
 use Symfony\Component\DependencyInjection\ChildDefinition;
 use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Exception\LogicException;
 
 /**
  * @author Alexander M. Turek <me@derrabus.de>
  */
-final class AttributeAutoconfigurationPass implements CompilerPassInterface
+final class AttributeAutoconfigurationPass extends AbstractRecursivePass
 {
     private $ignoreAttributesTag;
+    private $argumentConfigurators;
 
     public function __construct(string $ignoreAttributesTag = 'container.ignore_attributes')
     {
         $this->ignoreAttributesTag = $ignoreAttributesTag;
+
+        $this->argumentConfigurators = [
```

Comment:
We'd better remove this property, we don't need it, isn't it?

Reply:
This could be a local variable of `processDefinition()`, but that method will be called a lot, so I wanted to keep the array initialized.